subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/internal/DefaultJvmTestSuite.java

```diff
@@ -258,7 +268,20 @@ public void useJUnitJupiter(Provider<String> version) {
 
     @Override
     public void useSpock() {
-        useSpock(Frameworks.SPOCK.defaultVersion);
+        useSpockForDerivedGroovyVersion();
+    }
+
+    private void useSpockForDerivedGroovyVersion() {
+        Provider<VersionNumber> groovyVersionProvider = getProviderFactory().provider(() -> {
+            Optional<VersionNumber> maybeGroovyJarVersionNumber = runtimeClasspath.getFiles().stream()
```

Comment:
What do you think about using a `ResolutionResult` here instead of parsing class files?
```
        Optional<String> maybeGroovyVersion = runtimeClasspath.getIncoming().getResolutionResult().getAllComponents().stream()
            .map(ResolvedComponentResult::getModuleVersion)
            .filter(it -> ("org.codehaus.groovy".equals(it.getGroup()) || "org.apache.groovy".equals(it.getGroup())) && "groovy".equals(it.getName()))
            .map(ModuleVersionIdentifier::getVersion)
            .findFirst();
```

Reply:
If we didn't already have the `GroovyJarFile` utility class, this is the approach I would've taken.

I don't have strong feelings one way or the other. The disadvantage of `ResolutionResult` is that it results in a `String`, not a `VersionNumber` and also requires (yet another!) place we hardcode `"org.codehaus.groovy"` and `"org.apache.groovy"`.

Reply:
Personally, I don't think the `VersionNumber` part matters all that much, and there's no reason we can't encapsulate this logic in `GroovyJarFile` to keep it all centralized.
In general, I don't think it matters one way or another, though something about getting data from dependency-management vs parsing filenames seems a little more correct to me.

Reply:
Check out 1b64616199f07322062a67fc3c027f24d8fb1ea2. It's only one line longer and prevents any possibility of some component like `com.acme:groovy:1.2.3` from tripping-up `GroovyJarFile`. Matching on coordinates instead of filenames is good.

Reply:
Looks like GroovyJarFile also handles -all and -indy in the names. I think older versions of groovy supported a fat-jar based distribution. You may want to us `.startsWith("groovy")` instead of `equals`

Reply:
OK good point. We'll detect either `groovy` or `groovy-all`. `indy` is a classifier and not part of the name, so we can safely ignore it.